tests/lib/rules/no-shadow.js

```diff
@@ -91,6 +93,16 @@ eslintTester.addRuleTest("lib/rules/no-shadow", {
             errors: [{ message: "x is already declared in the upper scope.", type: "Identifier"}]
         },
         {
+            code: "{ let x = 2; } var x = 1;",
+            ecmaFeatures: {blockBindings: true},
+            errors: [{ message: "x is already declared in the upper scope.", type: "Identifier"}]
+        },
+        {
+            code: "function a() { let x = 2; } let x = 1;",
```

Comment:
This is not shadowing. Block variables are not hoisted, so inside `a` `x` is not yet defined. So this example shouldn't error.